src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Embedded.cs

```diff
@@ -0,0 +1,435 @@
+                // (4,18): error CS1729: 'object' does not contain a constructor that takes 0 arguments
+                //     public class Attribute {}
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "Attribute").WithArguments("object", "0").WithLocation(4, 18),
+                // (7,14): error CS1729: 'object' does not contain a constructor that takes 0 arguments
+                // public class Test
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "Test").WithArguments("object", "0").WithLocation(7, 14));
+        }
+
+        [Fact]
+        public void SynthesizingAttributeRequiresSystemAttribute_NoSystemVoid()
+        {
+            var code = @"
+namespace System
+{
+    public class Attribute {}
+    public class Object {}
+}
+public class Test
+{
+    public object M(ref readonly object x) { return x; } // should trigger synthesizing IsReadOnly
+}";
+
+            CreateCompilation(code).VerifyEmitDiagnostics(CodeAnalysis.Emit.EmitOptions.Default.WithRuntimeMetadataVersion("v4.0.30319"),
+                // (7,14): error CS0518: Predefined type 'System.Void' is not defined or imported
+                // public class Test
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "Test").WithArguments("System.Void").WithLocation(7, 14),
+                // (4,18): error CS0518: Predefined type 'System.Void' is not defined or imported
+                //     public class Attribute {}
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "Attribute").WithArguments("System.Void").WithLocation(4, 18),
+                // error CS0518: Predefined type 'System.Void' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Void").WithLocation(1, 1),
+                // error CS0518: Predefined type 'System.Void' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Void").WithLocation(1, 1));
+        }
+
+        [Fact]
+        public void SynthesizingAttributeRequiresSystemAttribute_NoDefaultConstructor()
+        {
+            var code = @"
+namespace System
+{
+    public class Object {}
+    public class Void {}
+    public class Attribute
+    {
+        public Attribute(object p) { }
+    }
+}
+public class Test
+{
+    public void M(ref readonly object x) { } // should trigger synthesizing IsReadOnly
+}";
+
+            CreateCompilation(code).VerifyEmitDiagnostics(CodeAnalysis.Emit.EmitOptions.Default.WithRuntimeMetadataVersion("v4.0.30319"),
+                // error CS1729: 'Attribute' does not contain a constructor that takes 0 arguments
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount).WithArguments("System.Attribute", "0").WithLocation(1, 1),
+                // error CS1729: 'Attribute' does not contain a constructor that takes 0 arguments
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount).WithArguments("System.Attribute", "0").WithLocation(1, 1));
```

Comment:
No. because the compiler is trying to synthesize two attributes here (`Embedded` and `IsReadOnly`).